File: ios/LusikSons/Theme/Haptics.swift
import UIKit

/// The app's haptic vocabulary — parity with the website's haptic.js
/// touchpoints (SiteProvider's add/remove buzzes, QuantityPicker's
/// steps), graded with native generators instead of navigator.vibrate's
/// single buzz. The system mutes these per user settings on its own;
/// haptics aren't motion, so no reduced-motion gate.
enum Haptics {
    /// Add to bag / buy now (web haptic(12)).
    static func add() {
        UIImpactFeedbackGenerator(style: .medium).impactOccurred()
    }

    /// Removing a bag row (web haptic(8)).
    static func remove() {
        UIImpactFeedbackGenerator(style: .light).impactOccurred()
    }

    /// Quantity / option stepping (web haptic(8) on the qty picker).
    static func step() {
        UISelectionFeedbackGenerator().selectionChanged()
    }

    /// Chrome taps — tab switches (Chunk 6's light impact, now shared).
    static func tap() {
        UIImpactFeedbackGenerator(style: .light).impactOccurred()
    }

    /// A completed moment — order placed, waitlist joined.
    static func success() {
        UINotificationFeedbackGenerator().notificationOccurred(.success)
    }
}

File: replit-app/src/lib/haptics.js
// ============================================================
// haptics — parity with ios/LusikSons/Theme/Haptics.swift (which
// itself mirrors the website's haptic.js touchpoints).
// ============================================================
// navigator.vibrate fires on Android browsers; iOS Safari has no
// vibration API and silently ignores it — visual feedback carries
// iPhones until WebKit ships one. Same vocabulary, same call sites
// as the Swift enum, so the two apps stay one design.

const buzz = (pattern) => {
  try {
    navigator?.vibrate?.(pattern);
  } catch {
    /* silent — not all devices support it */
  }
};

export const haptics = {
  /** Add to bag / buy now (Swift .add, web haptic(12)). */
  add: () => buzz(12),
  /** Removing a bag row (Swift .remove, web haptic(8)). */
  remove: () => buzz(8),
  /** Quantity / option stepping (Swift .step). */
  step: () => buzz(8),
  /** Chrome taps — tab switches (Swift .tap). */
  tap: () => buzz(6),
  /** A completed moment — order placed, waitlist joined (Swift .success). */
  success: () => buzz([10, 60, 14]),
};
